File: src/main/java/ca/joelathiessen/kaly2/subconscious/Measurement.java
package ca.joelathiessen.kaly2.subconscious;

import lejos.robotics.navigation.Pose;

public class Measurement {

    private float distance;
    private float angle;
    private Pose pose;
    private long time;

    public Measurement(float distance, float angle, Pose pose, long time) {
        this.distance = distance;
        this.angle = angle;
        this.pose = pose;
        this.time = time;
    }

    public float getDistance() {
        return this.distance;
    }

    public float getAngle() {
        return this.angle;
    }

    public Pose getPose() {
        return this.pose;
    }

    public long getTime() {
        return this.time;
    }

}

File: src/main/kotlin/ca/joelathiessen/kaly2/Measurement.kt
package ca.joelathiessen.kaly2

import ca.joelathiessen.util.FloatMath
import lejos.robotics.navigation.Pose

class Measurement(val distance: Float, val probAngle: Float, val probPose: Pose, val odoPose: Pose, val time: Long) {
    val dX = FloatMath.cos(probAngle) * distance
    val dY = FloatMath.sin(probAngle) * distance
    val x = probPose.x + dX
    val y = probPose.y + dY
}
